ERCS/erc-7562.md

```diff
@@ -0,0 +1,347 @@
+* Access to associated storage of the account in an external (non-entity contract) is allowed if either:
+    * **[STO-021]**  The account already exists
+    * **[STO-022]**  There is an `initCode` and the `factory` contract is staked.
+* If the entity (`paymaster`, `factory`) is staked, then it is also allowed:
+    * **[STO-031]** Access the entity's own storage.
+    * **[STO-032]** Read/Write Access to storage slots that is associated with the entity, in any non-entity contract.
+    * **[STO-033]** Read-only access to any storage in non-entity contract.
+* **[STO-040]** `UserOperation` may not use an entity address (`factory`/`paymaster`/`aggregator`) that is used as an "account" in another `UserOperation` in the mempool. \
+This means that a `Paymaster` and `Factory` contracts cannot practically be an "account" contract as well.
+* **[STO-041]** A contract whose Associated Storage slot is accessed during a `UserOperation` may not be an address of a "sender" in another `UserOperation` in the mempool.
+
+### Staked Entities Reputation Rules
+
+* **[SREP-010]** The "canonical mempool" defines a staked entity if it has `MIN_STAKE_VALUE` and unstake delay of `MIN_UNSTAKE_DELAY`
+* **[SREP-020]** A `BANNED` address is not allowed into the mempool.\
+  Also, all existing `UserOperations` referencing this address are removed from the mempool.
+* **[SREP-030]** A `THROTTLED` address is limited to:
+    * `THROTTLED_ENTITY_MEMPOOL_COUNT` entries in the mempool.
+    * `THROTTLED_ENTITY_BUNDLE_COUNT` `UserOperations` in a bundle.
+    * Can remain in the mempool only for `THROTTLED_ENTITY_LIVE_BLOCKS`.
+* **[SREP-040]** An `OK` staked entity is unlimited by the reputation rule.
+    * Allowed in unlimited numbers in the mempool.
+    * Allowed in unlimited numbers in a bundle.
+* **[SREP-050]** If a staked entity fails the second validation or fails bundle creation, its `opsSeen` is incremented by `10000`, causing it to be `BANNED`.
+
+### Entity-specific rules:
+
+* **[EREP-010]** For each `paymaster`, the mempool must maintain the total gas `UserOperations` using this `paymaster` may consume.
+    * Do not add a `UserOperation` to the mempool if the maximum total gas usage, including the new `UserOperation`, is above the deposit of the `paymaster` at the current gas price.
+    * TODO: allow "fractional reserve" based on reputation?
+* **[EREP-020]** A staked factory is "accountable" for account breaking the rules. \
+That is, if the `validateUserOp()` is rejected for any reason, it is treated as if the factory caused this failure, and thus this affects its reputation.
+* **[EREP-030]** A Staked Account is accountable for failures in other entities (`paymaster`, `aggregator`) even if they are staked.
+* **[EREP-040]** An `aggregator` must be staked, regardless of storage usage.
+* **[EREP-050]** An unstaked `paymaster` may not return a `context`.
+    * TODO: can be removed once we remove the "2nd postOp".
```

Comment:
```suggestion
<!--    * TODO: can be removed once we remove the "2nd postOp". -->
```

HTML comments are caught by the linter in Review and later stages, so it'll make sure these don't accidentally get left in.